### DISCUSSION TITLE
Cannot wrap custom component in next <Link>

### DISCUSSION THREAD

**Discussion by AdamZajler** [vercel/next.js#41076] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

**Operating System:**
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    
**Binaries:**
      Node: 16.17.1
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    
**Relevant packages:**
      next: 12.2.3
      eslint-config-next: 12.2.3
      react: 17.0.2
      react-dom: 17.0.2

### Describe the Bug

I have my own custom element named `<Typography>` it's full code is bellow (commented stuff is just a try to fix the issue): 

```
import React from 'react';

interface IComponentProps {
  component?: keyof JSX.IntrinsicElements;
  className?: string;
  children: React.ReactNode;
}

// const Typography = ({ component: Tag = 'p', className, children }: IComponentProps): JSX.Element => {
//   return <Tag className={className}>{children}</Tag>;
// };
const Typography = ({ component = 'p', className, children }: IComponentProps): JSX.Element => {
  return React.createElement(`${component}`, { className }, children);
};

export default Typography;
```

And in another element I'm trying to wrap `<Typography>` in next/link but in browser when I'm trying to click the text there is nothing happends 😢 and also in webtools this element `<h6>` have no `href` attribute.

```
      <Link href={'/artykuly/test'} passHref>
        <Typography component={'a'}>Artykuły</Typography>
        {/*<a>Artykuły</a>*/}
      </Link>
```

Only if i delete my custom element and use standard `<h6>` => then it have `href` attribute and is fully working

### Expected Behavior

My custom element will have href attribute & after I click it, it will transfer me to page

### Link to reproduction

https://codesandbox.io/s/wild-tree-cc3i2u?file=/pages/index.js

### To Reproduce

- create text using my custom element `Typography`
- wrap it around in `next/link <Link>`

**Comment by SukkaW** [comment]
This is because your custom component doesn't support passing a ref where `next/link` is trying to attach other extra props (like `onClick`) to it.

You should use the `forwardRef` in your custom component and pass the `ref` to the `<a />` inside your custom component.
